packages/js/components/src/experimental-select-control/hooks/use-async-filter.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { useDebounce } from '@wordpress/compose';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { SelectControlProps } from '../select-control';
+
+export const DEFAULT_DEBOUNCE_TIME = 250;
+
+export default function useAsyncFilter< T >( {
+	filter,
+	onFilterStart,
+	onFilterEnd,
+	onFilterError,
+	onInputChange,
```

Comment:
About the default values I'm partially agree with that. My reasoning about that is the following.
```typescript
function main(cb = () => {}) {
  cb();
}
```
if we call the prev function like `main()` or `main(undefinded)` the default value is gonna work but if for some reason we call `main(nullish_variable_not_undefined)` then the default value is not gonna work. Because of that I try to avoid default value attributes as much as possible.

What I described above has a close relation with what you said in your second point, about passing a non callable value, and of course, I'm in the same line as you. So this is defenately something I have to change to:
```typescript
if (typeof cb == 'function') cb();
```
the previous condition prevent the cb function of been called if it's anything else than just a function and also let us to not use default values.